src/services/upload/livePhotoService.ts

```diff
@@ -0,0 +1,162 @@
+import { FILE_TYPE } from 'constants/file';
+import { encodeMotionPhoto } from 'services/motionPhotoService';
+import {
+    FileTypeInfo,
+    FileWithCollection,
+    isDataStream,
+    LivePhotoAssets,
+    Metadata,
+} from 'types/upload';
+import { splitFilenameAndExtension } from 'utils/file';
+import { readFile } from './fileService';
+import uploadService from './uploadService';
+import UploadService from './uploadService';
+
+export function getLivePhotoFileType(
+    file1TypeInfo: FileTypeInfo,
+    file2TypeInfo: FileTypeInfo
+) {
+    return {
+        fileType: FILE_TYPE.LIVE_PHOTO,
+        exactType: `${file1TypeInfo.exactType}+${file2TypeInfo.exactType}`,
+    };
+}
+
+export function getLivePhotoMetadata(
+    file1Metadata: Metadata,
+    file2Metadata: Metadata
+) {
+    return {
+        ...file1Metadata,
+        ...file2Metadata,
+        title: `${splitFilenameAndExtension(file1Metadata.title)[0]}.zip`,
+        fileType: FILE_TYPE.LIVE_PHOTO,
+    };
+}
+
+export function getLivePhotoSize(livePhotoAssets: LivePhotoAssets) {
+    return livePhotoAssets.image.size + livePhotoAssets.video.size;
+}
+
+export async function readLivePhoto(
+    worker,
+    reader: FileReader,
+    fileTypeInfo: FileTypeInfo,
+    livePhotoAssets: LivePhotoAssets
+) {
+    const image = await readFile(
+        worker,
+        reader,
+        { exactType: fileTypeInfo.exactType, fileType: FILE_TYPE.IMAGE },
+        livePhotoAssets.image
+    );
+    const video = await readFile(
+        worker,
+        reader,
+        { exactType: fileTypeInfo.exactType, fileType: FILE_TYPE.VIDEO },
+        livePhotoAssets.video
+    );
+
+    if (isDataStream(video.filedata) || isDataStream(image.filedata)) {
+        throw new Error('too large live photo assets');
+    }
+    return {
+        filedata: await encodeMotionPhoto({
+            image: image.filedata as Uint8Array,
+            video: video.filedata as Uint8Array,
+            imageNameTitle: livePhotoAssets.image.name,
+            videoNameTitle: livePhotoAssets.video.name,
+        }),
+        thumbnail: video.hasStaticThumbnail ? video.thumbnail : image.thumbnail,
+        hasStaticThumbnail: !(
+            !video.hasStaticThumbnail || !image.hasStaticThumbnail
+        ),
+    };
+}
+
+export function clusterLivePhotoFiles(mediaFiles: FileWithCollection[]) {
+    const analysedMediaFiles: FileWithCollection[] = [];
+    mediaFiles.sort((media1Files, media2Files) =>
+        splitFilenameAndExtension(media1Files.file.name)[0].localeCompare(
+            splitFilenameAndExtension(media2Files.file.name)[0]
+        )
+    );
+    for (let i = 0; i < mediaFiles.length - 1; i += 2) {
+        const mediaFile1 = mediaFiles[i];
+        const mediaFile2 = mediaFiles[i + 1];
```

Comment:
Might be wrong but looks like this won't live photos correctly in cases like:
` normalFile.jpg, livephoto.jpg, livephoto.mov`.